chapters/11-version-control-workflow.Rmd

```diff
@@ -1,181 +1,600 @@
+make changes in a fork or branch and then request that those changes get merged
+into the main repository by the maintainer. Pull requests can come from forks
+or from separate branches from within the repository. They let maintainers and
+contributors review, discussion, request further changes, and approve the
+changes for being merged into the main branch. Think of pull requests like peer
+review of scientific work or collaborators reviewing a scientific report.
+
+Learning to effectively use these tools and concepts can make collaborating with 
+others much easier and makes it easier to contribute to other projects.
+You can even use this workflow as a solo developer. So what is this workflow?
+Figure TODO: REF and TODO: REF show the basic overview for workflows
+not using or using a fork-based approach. For non-forked-based workflows, the
+steps are to:
+
+1. Create a new branch.
+1. Make some edits to files and commit to the branch.
+1. When the changes are done and ready, submit a pull request for the new branch
+to be merged into the main branch.
+1. Pull request gets either accepts and merged or changes are requested, which
+require going back up to step 2.
+1. After pull request gets accepted and merged, the main branch now has the updated
+changes and the new branch can now be deleted.
+1. Process can repeat multiple times, in parallel or in sequence depending on the
+size of the team.
+
+TODO: Diagram needs to be fixed here, this is basic skeleton
+
+```{r fig-git-collaborative-workflow-no-forks, eval=FALSE}
+DiagrammeR::grViz("digraph {
+repo -> new_branch -> pr -> accepted -> repo
+}")
+```
 
-## Collaborating with Git
+For fork-based workflows, the steps are to:
+
+1. Create a fork of the main repository (if it hasn't been forked already).
+1. Optionally but strongly recommended, create a new branch in the forked repository.
+1. Make edits and changes to the files and commit to the branch.
+1. When ready, submit a pull request. Changes may be requested, which require going
+back to step 3.
+1. If the pull request gets accepted and merged, the main branch in the main
+repository is now updated and the new branch in the forked repository can be
+deleted.
+1. The forked repository now needs to be synchronized with the main repository.
+
+```{r fig-git-collaborative-workflow-forks, eval=FALSE}
+DiagrammeR::grViz("digraph {
+sami_repo -> amira_fork -> amira_new_branch -> pr_to_sami_repo -> 
+    pr_accepted -> sami_repo -> synch -> amira_fork
+}")
+```
 
-### Exercise: Consider ways of collaborating and their effectiveness.
+The main differences between using a non-forked-based approach compared to a
+forked-based approach is determined by whether you have access to modify ("write
+permissions") the main repository. Usually if you are in a team, you have
+permissions to modify and save ("write") changes to the main repository,
+so in that case you would use a non-forked-based approach.
```

Comment:
Depends on the project, team size, and competences I think. It can be a bit easier administratively to manage the project when using the branches rather than forks. But its really whatever the contributor feels comfortable with.